Strawpond charter for throwing pebbles at (#124)
operate-first/community: sig-community/charter.md

# SIG Community

## Scope

This SIG cares about all aspects of what is helpful and harmful to this open source software community.

> We understand that this project is a human endeavor, and it is our job to put people before practice, process, and even principle.

Reviewer: Nit: move this above **Scope** as kind of a mission statement

Author: Cool, thanks. I'm going to do this here in a split way so I don't outdate the PR by doing it in my own repo.

> We understand that this project is a human endeavor, and it is our job to put people before practice, process, and even principle.

Reviewer 2: What does it mean to "put people before principle"? If a principle is a fundamental belief held by the org, then isn't it a principle of the org to put people before practice and process? But, maybe I'm misunderstanding the meaning of "principle" here though.    

Maybe "and it is our job to put people before practice, process, and even progress". is an alternative?

Author: This is a nuanced reading, thank you. I didn't think of that recursive loop that might even cancel itself out, in a rules engine. I like your alternative, as it gets to the more radical community reliance and supportiveness I had in mind:

```suggestion
We understand that this project is a human endeavor, and it is our job to put people before practice, process, and even progress.
```

I do think there is a discussion to be had about if the principle of putting people first simple overrides other principles (like [Azimov's Three Laws of Robotics](https://en.wikipedia.org/wiki/Three_Laws_of_Robotics)?), but perhaps that's too nuanced for here and now? Or splitting too many hairs? It would like like this:

```suggestion
We understand that this project is a human endeavor, and it is our primary principle to put people before practice, process, and even other principles.
```

Author: Going with first suggestion, as per SIG Community meeting, philosophy for the future.

> # SIG Community
> 
> ## Scope
> 
> This SIG cares about all aspects of what is helpful and harmful to this open source software community.
> We understand that this project is a human endeavor, and it is our job to put people before practice, process, and even principle.

Author: ```suggestion
# SIG Community

This SIG cares about all aspects of what is helpful and harmful to this open source software community.
We understand that this project is a human endeavor, and it is our job to put people before practice, process, and even principle.

## Scope
```

**What does it mean to support a community?**
It means understanding and putting resources into whatever is important to people in this community.
We put our own time and energy into this support, and we help coordinate and direct project resources.

**What is an open source project compared to a community?**
A _project_ is a general term referring to an effort by people to accomplish something that requires time and multiple steps. Pruning a rose bush is a task, where a multi-year effort to plant a new rose garden is a project.
An open source software community is a type of _community of practice_, and the software project is the place where the community practices.

You can have a community without a project, which is really a social club.
But you cannot have an open source project without a community.
However, not all open source software is maintained by a project.

> It may rather be maintained by an individual, or a single organization by fiat.

Reviewer 3: Define 'fiat', as many people only know this definition of the term: https://en.wikipedia.org/wiki/Fiat

Author: > Define 'fiat', as many people only know this definition of the term: https://en.wikipedia.org/wiki/Fiat

Oh you are correct entirely, we need a different word instead of defining this one.

```suggestion
It may rather be maintained by an individual, or a single organization through its members (employees, students, researchers, etc.) being effectively the sole contributors.
```

Reviewer 3: Better. And then I guess that sentence is only a pre-amble.

### In scope

> 

Reviewer 3: Should we put as 1st item, that we want to establish a wide, diverse and inclusive community  from all backgrounds and vendors?

Author: ```suggestion

* Establishing a wide, diverse, and inclusive community of people and organizations from all backgrounds and situations.
```

* Creating and maintaining an open source governance for the project

>   - In particular establishing democratically elected overarching leadership or focus committees (Steering, Security, et al)

Reviewer 2: ```suggestion
  - In particular establishing democratically elected overarching leadership and focus committees (Steering, Security, et al)
```

* Establishing further SIGs to take over specific purviews from SIG Community
* Being widely inclusive and holding a vision for diversity and equity in the project
* Supporting diversity, equity, and inclusion with actions and resources
* Maintaining project Code of Conduct and related reporting and response processes
* Care about and improve the user and contributor experience
* Create and maintain user personas for the project
* Establish a project-wide documentation approach

> * Establish project-wide communication norms

Reviewer: ```suggestion
* Establish and maintain project-wide communication norms and channels
```

* User and contributor onboarding, i.e., the process for bringing new people into the project
* Creating and maintaining project role definitions and responsibility/authority matrix
* Keeping track of project and community health, including the use of metrics
* Work with SIG Operations to scope and define the infrastructure of participation for contributors
* Ownership and financing (domain names, other project assets)
* Ensure project-wide transparency

> * Set project technical direction and maintain a development roadmap

Reviewer 4: Should we be specific about communications and the website?

`* Care about the website presence and other communication outlets, e.g. social media and mail`

Author: ```suggestion
* Set project technical direction and maintain a development roadmap
* Care about the website presence and other communication outlets, e.g. social media and mail
```

### Out of scope

* Directing project infrastructure

> * Handling embargoed security discussions and responses

Reviewer 3: Does that mean that non-embargoed security discussions should be discussed in the Community SIG?  It feels a bit odd, but I can't explain

Author: Good clarification, thanks.

```suggestion
* Handling security discussions and responses, including having no role in embargoed security situations
```

> * Being the body that leads the project beyond the ratification of Governance 1.1, which will establish one or more leadership committees.

Reviewer: Can you please explain this to me? I do not understand it... Why is this out of scope and what it means?

Author: This is indeed pretty poorly written, thank you for flagging it.

The initial Governance 1.0 specifically calls out the SIG Community as responsible for seeing the work to openly discuss, create, and ratify the Governance 1.1 gets done.

This item is intended to clarify that once that part of that work is in Governance 1.1 a (probably) new body will be in charge of driving any future changes to the Governance. For example, one of the goals of Governance 1.1 is to move to a leadership committee of some kind, elected or selected in some way. Presumably that body will be assigned the responsibility of stewarding the Governance going forward.

There is also this implied effect of SIG Operations being responsible for only operating the cloud. That means the current responsibility to lead the project in its entirety sits with the SIG Community until the Gov 1.1 forms leadership committees to run and own things.

How does this change sound to your ear?
```suggestion
* Stewarding changes to the Governance beyond the work for Governance 1.1, after which one or more leadership committees will be accountable and responsible for stewarding beyond Gov 1.1.
* Being the de facto body leading the Op1st project once those capabilities and responsibilities are instilled into one or more leadership committees.
```

Author: Suggestion good enough for now

## Stakeholders

* Contributors, and by extension contributors' organizations
* OpenInfra Foundation, via OpenInfra Labs relationship

> * MOC?

Reviewer 4: I'm ok with naming MOC directly, since they're a key provider of key infra
cc @msdisme

Reviewer 3: Extend the MOC TLA

Author: ```suggestion
* [Mass Open Cloud (MOC)](https://massopen.cloud/)
```
